From d0790f438c549701d66d60f275e8e929ea78d410 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:30:06 -0400
Subject: [misc] genwqe: Ensure rc is not returning an uninitialized value

Message-id: <1413239416-7778-19-git-send-email-gduarte@redhat.com>
Patchwork-id: 97550
O-Subject: [RHEL7.1 PATCH BZ 1088491 18/28 v2] GenWQE: Ensure rc is not returning an uninitialized value
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: ebb2c96bb9214ba38c7fe35d5d725f6e7cb3bbc8

commit ebb2c96bb9214ba38c7fe35d5d725f6e7cb3bbc8
Author: Colin Ian King <colin.king@canonical.com>
Date:   Thu Mar 20 15:11:04 2014 +0100

    GenWQE: Ensure rc is not returning an uninitialized value

    rc is not initialized, so genwqe_finish_queue() either returns -EIO or
    garbage.  Fortunately the return is not being checked by any callers,
    so this has not yet caused any problems. Even so, it makes sense to
    fix this small bug in case is is checked in future.

    Signed-off-by: Colin Ian King <colin.king@canonical.com>
    Signed-off-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_ddcb.c b/drivers/misc/genwqe/card_ddcb.c
index 29a1a28..c8046db 100644
--- a/drivers/misc/genwqe/card_ddcb.c
+++ b/drivers/misc/genwqe/card_ddcb.c
@@ -1310,7 +1310,7 @@ static int queue_wake_up_all(struct genwqe_dev *cd)
  */
 int genwqe_finish_queue(struct genwqe_dev *cd)
 {
- int i, rc, in_flight;
+ int i, rc = 0, in_flight;
  int waitmax = genwqe_ddcb_software_timeout;
  struct pci_dev *pci_dev = cd->pci_dev;
  struct ddcb_queue *queue = &cd->queue;
-- 
1.7.1